conda/gateways/connection/session.py

```diff
@@ -107,6 +107,18 @@ def session_manager(url: str):
     return CondaSession(auth=auth_handler_cls(channel_name))
 
 
+def get_session_storage_key(auth) -> str:
+    """
+    Function that determines which storage key to use for our CondaSession object caching
+    """
+    if auth is None:
+        return "default"
+
+    auth_type = type(auth)
+
+    return f"{auth_type.__module__}.{auth_type.__qualname__}::{auth.channel_name}"
```

Comment:
Remember that in requests lore, auth can also be a tuple, this might break that contract.

Reply:
Yep, that should do the trick!